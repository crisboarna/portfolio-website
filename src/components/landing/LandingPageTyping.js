import React, { Component } from 'react';
import Typist from 'react-typist';

import '../../../node_modules/react-typist/dist/Typist.css';
import '../../styles/landing/LandingPageTyping.css';

export default class LandingPageTyping extends Component {
  constructor() {
    super();
    this.state = {
      typing: true,
      renderMsg: false
    }
  }

  done = () => {
    this.setState({ typing: false }, () => {
      this.setState({ typing: true })
    });
  };

  onIAmTypingDone = () => {
    this.setState({renderMsg:true});
  };

  render() {
    return this.state.typing
      ? <div className="row">
        <Typist className='col-xs-2 col-sm-offset-4 typing-top landing-typing'
        avgTypingDelay={100}
        cursor={{hideWhenDone:true, hideWhenDoneDelay:0}}
        onTypingDone={this.onIAmTypingDone}>
          <b className='pulsate'>I am</b>
        </Typist>
        {this.state.renderMsg ?
        <Typist
          className='col-md-6 col-xs-pull-1 typing landing-typing'
          avgTypingDelay={100}
          startDelay={1000}
          onTypingDone={this.done}>
          creative.
          <Typist.Delay ms={500} />
          <Typist.Backspace count={9} delay={200} />
          the enabler of your dreams.
          <Typist.Delay ms={500} />
          <Typist.Backspace count={27} delay={200} />
          a fullstack developer.
          <Typist.Delay ms={1500} />
          <Typist.Backspace count={24} delay={200} />
          Cristian.
          <Typist.Delay ms={3000} />
        </Typist> : ''}
      </div>
      : ''
  }
}