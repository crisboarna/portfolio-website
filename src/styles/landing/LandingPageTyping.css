.landing-typing {
    position:absolute;
    top:50%;
}

.typing {
    padding-left:5%;
    font-size: 70px;
    letter-spacing: 1px;
    color: #f8f8f8;
    text-shadow: -1px 0 black, 0 1px black, 1px 0 black, 0 -1px black;
    -webkit-animation: secondTypingEffect 0.5s forwards;
    animation: secondTypingEffect 0.5s forwards;}

@keyframes secondTypingEffect {
    0%   {opacity: 0;}
    90%  {opacity: 0;}
    100% {opacity: 1;}
}
@-webkit-keyframes secondTypingEffect {
    0%   {opacity: 0;}
    90%  {opacity: 0;}
    100% {opacity: 1;}
}

.typing-top {
    padding-left:5%;
    font-size: 70px;
    letter-spacing: 1px;
    color: #f8f8f8;
    text-shadow: -1px 0 black, 0 1px black, 1px 0 black, 0 -1px black;
}

.pulsate {
    -webkit-animation: pulsate 2s ease-out;
    -webkit-animation-iteration-count: infinite;
    opacity: 0.5;
    color: #27a9ff;
}
@-webkit-keyframes pulsate {
    0% {
        opacity: 0.5;
    }
    50% {
        opacity: 1.0;
    }
    100% {
        opacity: 0.5;
    }
}

@media all and (max-width: 660px) {
    .typing {
        position: absolute;
        top: 50%;
        width: 100%;
        font-size: 70px;
        letter-spacing: 1px;
        color: #f8f8f8;
        text-shadow: -1px 0 black, 0 1px black, 1px 0 black, 0 -1px black;
    }
    .typing-top {
        position: absolute;
        top: 35%;
        width: 100%;
        font-size: 70px;
        letter-spacing: 1px;
        color: #f8f8f8;
        text-shadow: -1px 0 black, 0 1px black, 1px 0 black, 0 -1px black;
    }
}