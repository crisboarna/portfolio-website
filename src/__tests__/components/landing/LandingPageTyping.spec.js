import React from 'react';
import { shallow, mount } from 'enzyme';
import LandingPageTyping from '../../../components/landing/LandingPageTyping';

describe('LandingPageTyping', () => {
  it('renders without crashing', () => {
    const component = shallow(<LandingPageTyping />);
    expect(component.exists()).toEqual(true);
  });

  it('compare to snapshot', () => {
    const component = mount(<LandingPageTyping/>);
    expect(component).toMatchSnapshot();
  });

  it('sets state to message rendered', () => {
    const component = shallow(<LandingPageTyping/>);
    component.instance().onIAmTypingDone();
    expect(component.state().renderMsg).toEqual(true);
  });

  it('sets state to typing done', () => {
    const component = shallow(<LandingPageTyping/>);
    component.instance().done();
    expect(component.state().typing).toEqual(true);
  });
});