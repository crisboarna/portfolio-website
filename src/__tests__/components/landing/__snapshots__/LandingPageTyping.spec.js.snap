// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPageTyping compare to snapshot 1`] = `
<LandingPageTyping>
  <div
    className="row"
  >
    <Typist
      avgTypingDelay={100}
      className="col-xs-2 col-sm-offset-4 typing-top landing-typing"
      cursor={
        Object {
          "hideWhenDone": true,
          "hideWhenDoneDelay": 0,
        }
      }
      delayGenerator={[Function]}
      onCharacterTyped={[Function]}
      onLineTyped={[Function]}
      onTypingDone={[Function]}
      startDelay={0}
      stdTypingDelay={25}
    >
      <div
        className="Typist col-xs-2 col-sm-offset-4 typing-top landing-typing"
      >
        <Cursor
          blink={true}
          element="|"
          hideWhenDone={true}
          hideWhenDoneDelay={0}
          isDone={false}
          show={true}
        >
          <span
            className="Cursor Cursor--blinking"
          >
            |
          </span>
        </Cursor>
      </div>
    </Typist>
  </div>
</LandingPageTyping>
`;
